src/circuit_writer/mod.rs

```diff
@@ -150,11 +150,13 @@ impl<B: Backend> CircuitWriter<B> {
         let qualified = FullyQualified::local("main".to_string());
         let main_fn_info = circuit_writer.main_info()?;
 
-        let function = match &main_fn_info.kind {
+        let mut function = match &main_fn_info.kind {
             crate::imports::FnKind::BuiltIn(_, _) => unreachable!(),
             crate::imports::FnKind::Native(fn_sig) => fn_sig.clone(),
         };
```

Comment:
maybe clearer to write something like this:

```rust
// handle public inputs in order
for arg in function.sig.arguments.iter().filter(|arg| arg.is_public()) {
  handle_arg(arg, circuit_writer.add_public_inputs)?;
}

// handle private inputs in order
for arg in function.sig.arguments.iter().filter(|arg| arg.is_private())
  handle_arg(arg, circuit_writer.add_private_inputs)?;
}
```

which makes the handling of public inputs before private inputs much more explicit

Reply:
Refactored: [99f328f](https://github.com/zksecurity/noname/pull/78/commits/99f328fa9d6799148eff5dd65a699758aeeceebe)

Thanks for the advice.